sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -89,9 +89,10 @@ class RegexParser(pattern: String) {
     }
   }
 
-  private def parseFactor(): RegexAST = {
+  private def parseFactor(until: () => Boolean): RegexAST = {
     var base = parseBase()
-    while (!eof() && (peek().exists(ch => ch == '*' || ch == '+' || ch == '?')
+    while (!eof() && !until()
+        && (peek().exists(ch => ch == '*' || ch == '+' || ch == '?')
```

Comment:
Per previous nit, should be able to simplify this to something like:
```suggestion
    while (!until() && (peek().exists(ch => ch == '*' || ch == '+' || ch == '?')
```